### ISSUE TITLE
[TG LLama-70B] Add device sampling support to vLLM demo

### ISSUE THREAD

**Issue by mtairum** [tenstorrent/tt-metal#20181]
Currently the 70B demo is running sampling on host.

As the decode team is porting sampling to device, we require the generator to also support this in `models/demos/llama3_subdevices/tt/generator.py`.

Initial assignment to @skhorasganiTT. Let me know if you need help with this task.

Some questions:
- Where does vLLM call sampling? Is it through the sample generator?
  - If so can we just switch for the on-device sampling?
- On the decode trace call inside the generator, do we want to return logits or do sampling inside this call and return the tokens?

TODO:
- Provide device sampling test code.

Extra goal: Port sampling support for TT-Transformers (requires new ticket to track this).

**Comment by skhorasganiTT** [comment]
I have added support in vLLM for the option to pass in sampling args to TT models in https://github.com/tenstorrent/vllm/pull/76. Also, modified TT-Transformers to work with this for the batch-1 + argmax case in https://github.com/tenstorrent/tt-metal/pull/20458.
I believe this should unblock TG-Llama for sampling on device in vLLM,  @sraizada-tt can you try it out by modifying your Generator's decode forward function to accept the sampling params and do sampling/argmax on device?
